machine/v1alpha1/provider_types.go

```diff
@@ -0,0 +1,207 @@
+	// https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases.
+	// +optional
+	FetchConfig *FetchConfiguration `json:"fetchConfig,omitempty"`
+}
+
+// ManagerSpec defines the properties that can be enabled on the controller manager for the provider.
+type ManagerSpec struct {
+	// ControllerManagerConfigurationSpec defines the desired state of GenericControllerManagerConfiguration.
+	ctrlconfigv1.ControllerManagerConfigurationSpec `json:",inline"`
+
+	// ProfilerAddress defines the bind address to expose the pprof profiler (e.g. localhost:6060).
+	// Default empty, meaning the profiler is disabled.
+	// Controller Manager flag is --profiler-address.
+	// +optional
+	ProfilerAddress *string `json:"profilerAddress,omitempty"`
+
+	// MaxConcurrentReconciles is the maximum number of concurrent Reconciles
+	// which can be run. Defaults to 10.
+	// +optional
+	// +kubebuilder:default=10
+	// +kubebuilder:validation:Minimum=1
+	MaxConcurrentReconciles *int `json:"maxConcurrentReconciles,omitempty"`
+
+	// Verbosity set the logs verbosity. Defaults to 1.
+	// Controller Manager flag is --verbosity.
+	// +optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	Verbosity int `json:"verbosity,omitempty"`
+
+	// Debug, if set, will override a set of fields with opinionated values for
+	// a debugging session. (Verbosity=5, ProfilerAddress=localhost:6060)
+	// +optional
+	// +kubebuilder:default=false
+	Debug bool `json:"debug,omitempty"`
+
+	// FeatureGates define provider specific feature flags that will be passed
+	// in as container args to the provider's controller manager.
+	// Controller Manager flag is --feature-gates.
+	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+}
+
+// DeploymentSpec defines the properties that can be enabled on the Deployment for the provider.
+type DeploymentSpec struct {
+	// Number of desired pods. This is a pointer to distinguish between explicit zero and not specified. Defaults to 1.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	Replicas *int `json:"replicas,omitempty"`
+
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// If specified, the pod's tolerations.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// If specified, the pod's scheduling constraints
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+
+	// List of containers specified in the Deployment
+	// +optional
+	Containers []ContainerSpec `json:"containers"`
+}
+
+// ContainerSpec defines the properties available to override for each
+// container in a provider deployment such as Image and Args to the container’s
+// entrypoint.
+type ContainerSpec struct {
```

Comment:
this looks like a partial podspec?  What PodSpec does it correspond to so I can decide what I want to upgrade.